db/controllers/articles.controller.js

```diff
@@ -0,0 +1,15 @@
+const { fetchArticleById } = require('../models/articles.model.js')
+
+exports.getArticleById = (req, res, next) => {
+    const articleId = +req.params.article_id;
+    if (isNaN(articleId)){
+        return res.status(400).send({ message: "ID must be a number" });
```

Comment:
This error handling is fine for an invalid Id, but psql is expecting a number anyway! If it doesn't get one it will throw it's own error. You have already built the psql handler for this, so it might just be a case of linking it up in your controller, with a catch block! 👍